### DISCUSSION TITLE
Migration from Fluent journal files

### DISCUSSION THREAD

**Discussion by rploconnor** [ansys/pyfluent#71] [Ideas]
Are there any plans for a feature which translates journal commands to Python commands? This could help with the eventual migration to PyFluent. If there were a translator, ideally it wouldn't just pass a string containing the journal commands as the argument to some function.

**Comment by dnwillia-work** [comment]
Hi Ryan, not for 2022 R2.  For now the TUI scripts you have on hand must be migrated to use the `session.tui.x.y.z` API calls.  It should be a matter of adjusting the scripts you have to use the dot syntax and prefix all the commands with `session.tui`.

**Reply by rploconnor** [reply]
Okay, sounds good. Thanks!

**Comment by NSchmidtRose** [comment]
We are at 23R2 now, and the feature was apparently introduced in 23R1. Nevertheless, all I can find on it is that it indeed exists [e.g. in the "What's new" Video for 23R1 (is this the same Ryan O'Connor, btw?)], but not where to find it. Could anyone please enlighten me?

**Reply by mkundu1** [reply]
PyFluent's [launch_fluent](https://fluent.docs.pyansys.com/version/stable/api/general/launcher/index.html#ansys.fluent.core.launcher.launcher.launch_fluent) function has a `topy` option to convert Fluent journal(s) to Python journal.

The Fluent side feature is documented in the section "35.4.6. Automating Python Journals" in Fluent's beta documentation.
